avro-converter/src/test/java/io/confluent/connect/avro/AvroDataTest.java

```diff
@@ -566,6 +566,19 @@ public void testToConnectArray() {
   }
 
   @Test
+  public void testToConnectOptionalArray() {
+    org.apache.avro.Schema avroSchema = org.apache.avro.SchemaBuilder.builder()
+        .unionOf().nullType().and()
+        .array().items().intType().endUnion();
+    avroSchema.getElementType().addProp("connect.type", "int8");
+    // Use a value type which ensures we test conversion of elements. int8 requires extra
+    // conversion steps but keeps the test simple.
+    Schema schema = SchemaBuilder.array(Schema.OPTIONAL_INT8_SCHEMA).build();
```

Comment:
@bcotton It doesn't look like these schemas match up. The Avro schema you built is a union of [null, array of int]. The Connect schema is an array of optional int8, which translates to array of union of [null, int8] in Avro.

In fact, the cause of the failure currently is an exception on the line adding the `"connect.type"` to `"int8"` mapping because the avro schema is a union, so `getElementType()` is not valid.

I'm not sure of the exact scenario you are trying to test, but the follow is a variant where the schemas are consistent and match what I think you intended for the Avro type, and it passes:

```
    org.apache.avro.Schema avroSchema = org.apache.avro.SchemaBuilder.builder()
        .unionOf().nullType().and()
        .array().items().intType().endUnion();
    avroSchema.getTypes().get(1).getElementType().addProp("connect.type", "int8");
    // Use a value type which ensures we test conversion of elements. int8 requires extra
    // conversion steps but keeps the test simple.
    Schema schema = SchemaBuilder.array(Schema.INT8_SCHEMA).optional().build();
    assertEquals(new SchemaAndValue(schema, Arrays.asList((byte) 12, (byte) 13)),
                 avroData.toConnectData(avroSchema, Arrays.asList(12, 13)));
```